File: optimizer-demo.cs
using iText.License;

using iText.Pdfoptimizer;
using iText.Pdfoptimizer.Handlers;
using iText.Pdfoptimizer.Handlers.Converters;
using iText.Pdfoptimizer.Handlers.Imagequality.Processors;

using System.IO;

namespace PdfOptimizer_Demo_Lab
{
    class PdfOptimizer_Demo
    {
        public static string ORIG = "/uploads/input.pdf";
        public static string OUTPUT_FILE = "/myfiles/output.pdf";

        public static float compression_level = 0.5f;
        public static float image_scalar_level = 0.5f;

        public static void Main()
        {
            PdfOptimizer_Demo test = new PdfOptimizer_Demo();
            test.PdfOptimizerDemo();
        }

        public void PdfOptimizerDemo()
        {

            PdfOptimizer optimizer = new PdfOptimizer();

            optimizer.AddOptimizationHandler(new FontDuplicationOptimizer());

            optimizer.AddOptimizationHandler(new FontSubsettingOptimizer());

            ImageQualityOptimizer tiff_optimizer = new ImageQualityOptimizer();
            tiff_optimizer.SetTiffProcessor(new BitmapCompressor(image_scalar_level, compression_level));
            optimizer.AddOptimizationHandler(new ImageQualityOptimizer());

            optimizer.AddOptimizationHandler(new CompressionOptimizer());

            ColorSpaceConverter RGB_to_CMYK_Converter = new ColorSpaceConverter();
            CsConverterProperties csConversionProperties = new CsConverterProperties(ColorConversionMode.NORMAL);
            RGB_to_CMYK_Converter.SetCsConverter(new RgbToCmykCsConverter(csConversionProperties));
            optimizer.AddOptimizationHandler(RGB_to_CMYK_Converter);

            optimizer.Optimize(
                    new FileInfo(ORIG, FileMode.Open),
                    new FileInfo(OUTPUT_FILE, FileMode.Create, FileAccess.Write));
        }
    }
}

File: optimizer-demo.java
import com.itextpdf.licensekey.LicenseKey;

import com.itextpdf.pdfoptimizer.PdfOptimizer;
import com.itextpdf.pdfoptimizer.handlers.ColorSpaceConverter;
import com.itextpdf.pdfoptimizer.handlers.CompressionOptimizer;
import com.itextpdf.pdfoptimizer.handlers.FontDuplicationOptimizer;
import com.itextpdf.pdfoptimizer.handlers.FontSubsettingOptimizer;
import com.itextpdf.pdfoptimizer.handlers.ImageQualityOptimizer;
import com.itextpdf.pdfoptimizer.handlers.converters.ColorConversionMode;
import com.itextpdf.pdfoptimizer.handlers.converters.CsConverterProperties;
import com.itextpdf.pdfoptimizer.handlers.converters.RgbToCmykCsConverter;
import com.itextpdf.pdfoptimizer.handlers.imagequality.processors.BitmapCompressor;

import java.io.File;
import java.io.IOException;

public class PdfOptimizer_Demo_Lab {

   private static final String ORIG = "/uploads/optimize.pdf";
   private static final String OUTPUT_FOLDER = "/myfiles/output_optimized.pdf";

   public static float compression_level = 0.5f;
   public static float image_scalar_level = 0.5f;

    public static void main(String args[]) throws IOException {
        PdfOptimizer_Demo_Lab demo = new PdfOptimizer_Demo_Lab();
        demo.pdfOptimizerDemo();
    }

    public void pdfOptimizerDemo() throws IOException {

        PdfOptimizer optimizer = new PdfOptimizer();

        optimizer.addOptimizationHandler(new FontDuplicationOptimizer());
        
        optimizer.addOptimizationHandler(new FontSubsettingOptimizer());

        ImageQualityOptimizer tiff_optimizer = new ImageQualityOptimizer();
        tiff_optimizer.setTiffProcessor(new BitmapCompressor(image_scalar_level, compression_level));
        optimizer.addOptimizationHandler(new ImageQualityOptimizer());

        optimizer.addOptimizationHandler(new CompressionOptimizer());
        
        ColorSpaceConverter RGB_to_CMYK_Converter = new ColorSpaceConverter();
        CsConverterProperties csConversionProperties = new CsConverterProperties(ColorConversionMode.NORMAL);
        RGB_to_CMYK_Converter.setCsConverter(new RgbToCmykCsConverter(csConversionProperties));
        optimizer.addOptimizationHandler(RGB_to_CMYK_Converter);

        optimizer.optimize(
                new File(ORIG),
                new File(OUTPUT_FOLDER));
    }
}
